src/uinputevents.h

```diff
@@ -0,0 +1,51 @@
+// This file is part of Projecteur - https://github.com/jahnf/projecteur - See LICENSE.md and README.md
+// Virtal Device to emit customized events from Projecteur device
+// The spotlight.cc grabs mouse inputs from Logitech Spotlight device.
+// This module is used when the input events are supposed to be forwarded to the system.
+# pragma once
+
+#include <iostream>
+#include <memory>
+#include <linux/uinput.h>
+#include <unistd.h>
+
+#include <QDebug>
+
+using namespace std;
+
+class uinputEvents;
+
+class uinputEvents{
+  private:
+    // Device that can act as virtual keyboard and mouse
+    struct uinput_user_dev uinp;
+	static int uinp_fd;
+
+    uinputEvents(){
+      uinp_fd = -1;
+      // Try to setup the device. If it fails exit.
+      if (setup_uinputDevice() != 1)
+         exit(1);
```

Comment:
If no `uinputDevice` is setup then we cannot grab the SpotLight device. Without grabbing the `SpotLight` device, it is impossible to provide the feature of mode change.

Further, I guess if user is not able to make a `uinput` device then it is major problem with System setup. A user on properly configured system should be able to make `uinput` device so this will not fail (without extra privilege; `uinput` is supposed to be used by userland software).

Reply:
It should not be the decision of the uinputevent class to just exit the program. This is basically like a crash, without a proper clean up.

Reply:
If the program is not able to make a virtual device, then it notifies the user and work as expected (without the features that rely on grabbing SpotLight device).